### PR TITLE
[Order] Use typehints introduced by PHP7+

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | part of #8425 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Component/Order/Model/AdjustmentInterface.php [hunk 1]
```diff
@@ -22,85 +22,85 @@
...
+     * @return AdjustableInterface|null
      */
-    public function getAdjustable();
+    public function getAdjustable(): ?AdjustableInterface;
 
     /**
      * @param AdjustableInterface|null $adjustable
      */
-    public function setAdjustable(AdjustableInterface $adjustable = null);
+    public function setAdjustable(?AdjustableInterface $adjustable): void;
```
Comment (Zales0123) [inline src/Sylius/Component/Order/Model/AdjustmentInterface.php]: We should mention this in `UPGRADE.md` file

File: src/Sylius/Component/Order/Model/OrderAwareInterface.php [hunk 2]
```diff
@@ -19,12 +19,12 @@
...
      */
-    public function getOrder();
+    public function getOrder(): ?OrderInterface;
 
     /**
-     * @param OrderInterface $order
+     * @param OrderInterface|null $order
      */
-    public function setOrder(OrderInterface $order = null);
+    public function setOrder(?OrderInterface $order): void;
```
Comment (Zales0123) [inline src/Sylius/Component/Order/Model/OrderAwareInterface.php]: And this.

File: src/Sylius/Component/Order/Model/OrderInterface.php [hunk 3]
```diff
@@ -28,119 +28,119 @@
...
      */
-    public function getCheckoutCompletedAt();
+    public function getCheckoutCompletedAt(): ?\DateTimeInterface;
 
     /**
-     * @param null|\DateTimeInterface $checkoutCompletedAt
+     * @param \DateTimeInterface|null $checkoutCompletedAt
      */
-    public function setCheckoutCompletedAt(\DateTimeInterface $checkoutCompletedAt = null);
+    public function setCheckoutCompletedAt(?\DateTimeInterface $checkoutCompletedAt): void;
```
Comment (Zales0123) [inline src/Sylius/Component/Order/Model/OrderInterface.php]: And this and everything else where we remove the default value from public method.

**Comment by Zales0123** [comment]
Thank you Kamil! 🥇

**Comment by pamil** [comment]
Thank you for this medal, it means so much for me!

**Pull request merged**